### PR TITLE
Feature/41191

### PR DESCRIPTION
**Pull request by UrvashiChaudhary**
Add following new props to the table component:
1. actions: Optional, list of actions to be rendered along with their handlers
2. count: No. of total records (Numeric), if count is 0 then don’t show the pagination.
3. page: Current Page, default to 0
4. rowsPerPage: No. of records per page, default to 100
5. onChangePage: handler to be called and update the page in the state.

In actions provide two actions i.e Edit and Remove,
- on Edit Handler, open the Edit Dialog prefilled with name and email, and in case of submit, show the the update data, in the “TraineeList” component,
- while for Remove Handler show the confirm Dialog Box, with cancel and Delete option. If Delete is clicked, then show the deleted item details in the console

Please refer to the attached GIFs.

Please note two more components need to be created adjacent to “AddDialog”, and they named as “EditDialog” and “RemoveDialog” respectively.

### CODE REVIEW

File: training/src/pages/Trainee/TraineeList.jsx [hunk 1]
```diff
@@ -55,8 +63,61 @@ class TraineeList extends React.Component {
     });
   };
 
+  handleChangePage = (event, newPage) => {
```
Comment (adilsuccessive) [inline training/src/pages/Trainee/TraineeList.jsx:65]: event is not used.

File: training/src/pages/Trainee/TraineeList.jsx [hunk 2]
```diff
@@ -55,8 +63,61 @@ class TraineeList extends React.Component {
...
+    this.setState({
+      page: newPage,
+    });
+  };
+
+  // eslint-disable-next-line no-unused-vars
+  handleRemoveDialogOpen = (element) => (event) => {
```
Comment (adilsuccessive) [inline training/src/pages/Trainee/TraineeList.jsx]: event is not used

File: training/src/pages/Trainee/TraineeList.jsx [hunk 3]
```diff
@@ -55,8 +63,61 @@ class TraineeList extends React.Component {
...
+    const { deleteData } = this.state;
+    this.setState({
+      RemoveOpen: false,
+    });
+    // eslint-disable-next-line no-console
+    console.log('Deleted Item ', deleteData);
+  };
+
+  // eslint-disable-next-line no-unused-vars
+  handleEditDialogOpen = (element) => (event) => {
```
Comment (adilsuccessive) [inline training/src/pages/Trainee/TraineeList.jsx:95]: event is not used

**Review by adilsuccessive — CHANGES_REQUESTED** [review]
if count is 0 then don’t show the pagination.
is not working

File: training/src/pages/Trainee/TraineeList.jsx (hunk 1 above)
Comment (UrvashiChaudhary) [reply training/src/pages/Trainee/TraineeList.jsx:65]: done

File: training/src/pages/Trainee/TraineeList.jsx (hunk 2 above)
Comment (UrvashiChaudhary) [reply training/src/pages/Trainee/TraineeList.jsx]: done

File: training/src/pages/Trainee/TraineeList.jsx (hunk 3 above)
Comment (UrvashiChaudhary) [reply training/src/pages/Trainee/TraineeList.jsx:95]: done

**Comment by UrvashiChaudhary** [comment]
> if count is 0 then don’t show the pagination.
> is not working

done

**Review by adilsuccessive — APPROVED** [review]

**Pull request merged**